javacutil/src/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -403,8 +411,21 @@ public static Tree getAssignmentContext(final TreePath treePath) {
                 }
                 // Otherwise use the context of the ConditionalExpressionTree.
                 return getAssignmentContext(parentPath);
-            case ASSIGNMENT:
             case METHOD_INVOCATION:
+                MethodInvocationTree methodInvocation = (MethodInvocationTree) parent;
+                // This was copied from old code.  Probably can be removed.
+                if (methodInvocation.getMethodSelect().getKind() == Kind.MEMBER_SELECT
+                        && ((MemberSelectTree) methodInvocation.getMethodSelect()).getExpression()
+                                == treePath.getLeaf()) {
+                    // a().b()
+                    // a() is the leaf of TreePath.
+                    // parent is a().b().
+                    return null;
```

Comment:
> For a().b() shouldn't the assignment context for call a() be the receiver type of b()?

No, the type of `a()` doesn't depend on the type of `b()`, because `a()` is used to determine which method is invoked by `b()`.

>Why is null useful?

`null` means no assignment context.

Reply:
I agree that for the Java type, the receiver type of `b()` doesn't matter.
But for the type annotations, wouldn't it be useful? If the call to `a()` allows any suitable type, but the receiver of `b()` adds a constraint, inference would fail if this isn't used as assignment context.